proj/src/main/webapp/script.js

```diff
@@ -38,28 +35,34 @@ function appendMessage(payload){
 /*
     Realtime Database
  */
+
 const path = '/messages'; // can make this more detailed (for example add user ID)
+const LIMIT = 20; // how many messages to load at a time
+var FIRST_CHILD_KEY;
 
 function init() {
     initRef();
+
+    const chat = document.getElementById('chat-as-list');
+    chat.addEventListener('scroll', addMoreMessagesAtTheTop);
```

Comment:
Interesting. I didn't know there was a scroll event listener. Does this happen every time you scroll or only when you scroll to the top? What if they are trying to scroll down even though they are the bottom?

Reply:
it only happens when you scroll to the top, i use an if statement to check if they're at the top